src/modules/control.cpp

```diff
@@ -34,13 +34,70 @@ static uint16_t targetVolume;
 static uint32_t totalBreathTime;
 static uint32_t targetInhalationTime;
 
+// Added
+static uint32_t targetExhalationTime;
+
+// Added
+static uint16_t targetPressure;
+
 static uint32_t measuredInhalationTime;
 static uint32_t measuredExhalationTime;
 
+// Control variables
+static uint16_t motorCommandPosition;		// Commands to low-level controller
+static uint16_t motorCommandVelocity;
+
+static uint16_t desiredVolume;				// Where we set desired volume to reach
+static uint16_t desiredPressire;			// " " ...
+static uint16_t desiredFlow;
+static uint16_t currentVolume;				// Need to update volume measurement here from sensors
+static uint16_t currentPressure;			// " " ...
+static uint16_t currentFlow;
+
 // Mid-level volume control function
 static void updateVolumeControl(uint16_t* distance, uint16_t* duration)
 {
-  // TODO: fill-in this function
+	// Control parameters
+	uint16_t volumeToPosition = 1;				// Relationship between volume and motor position
+	uint16_t pressureToVelocity = 1;		// Relationship between pressure and motor velocity
+	uint16_t pressureDerivGain = 1;			// Derivative gain that controls pressure tracking smoothness
+
+	// Check control mode
+	if (parameters.ventilationMode == VENTILATOR_MODE_VC) {
+		// Check state
+		if (control.state == CONTROL_INHALATION) {
+			// Check if reached volume set point
+			desiredVolume = targetVolume;
+			if (abs(currentVolume - desiredVolume) > 0) {
```

Comment:
Might want to add a margin of error to this, its unlikely wed hit exactly 0 wrt to the sensitivity of the volume measurement

Reply:
Good input! Will add this.